demo/midgard-node/src/transactions/utils.ts

```diff
@@ -167,3 +176,27 @@ export const outRefsAreEqual = (outRef0: OutRef, outRef1: OutRef): boolean => {
     outRef0.outputIndex === outRef1.outputIndex
   );
 };
+
+export class TransactionError extends Data.TaggedError("TransactionError")<{
+  readonly message: string;
+  readonly operation: "sign" | "submit" | "confirm";
+  readonly txHash?: string;
+  readonly cause?: unknown;
+}> {
```

Comment:
Did you consider defining 3 separate classes? e.g. `TxSignError`, `TxSubmitError`, and `TxConfirmError`

This way we'd be relieved from `switch` syntax, and utilize `effect`'s `catchTag`.